core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1087,7 +1087,9 @@ object GroupMetadataManager {
     val version =
       if (metadataVersion.isLessThan(IBP_2_1_IV0) || offsetAndMetadata.expireTimestamp.nonEmpty) 1.toShort
       else if (metadataVersion.isLessThan(IBP_2_1_IV1)) 2.toShort
-      else 3.toShort
+      // Serialize with the highest supported non-flexible version
+      // until a tagged field is introduced or the version is bumped.
+      else OffsetCommitValue.HIGHEST_SUPPORTED_NON_FLEXIBLE_VERSION
```

Comment:
nit: I would rather prefer to keep using the explicit version here. We could have an issue with `HIGHEST_SUPPORTED_NON_FLEXIBLE_VERSION` and that would mess things up.